File: app/src/main/java/com/example/mike/mp3player/service/library/search/SearchDatabase.java
package com.example.mike.mp3player.service.library.search;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Song.class, Folder.class}, version = 2)
public abstract class SearchDatabase extends RoomDatabase {
    public abstract SongDao songDao();
    public abstract FolderDao folderDao();

    private static final String DATABASE_NAME = "normalised_search_db";

    private static SearchDatabase instance;

    public synchronized static SearchDatabase getDatabase(final Context context) {
        if (instance == null) {
                     instance = Room.databaseBuilder(context.getApplicationContext(),
                            SearchDatabase.class, DATABASE_NAME)
                            .build();
                }
        return instance;
    }
}

File: service/src/main/java/com/github/goldy1992/mp3player/service/library/data/search/SearchDatabase.kt
package com.github.goldy1992.mp3player.service.library.data.search

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Song::class, Folder::class, Album::class], version = 3)
abstract class SearchDatabase : RoomDatabase() {
    abstract fun songDao(): SongDao
    abstract fun folderDao(): FolderDao
    abstract fun albumDao(): AlbumDao

    companion object {
        private const val DATABASE_NAME = "normalised_search_db"
        private lateinit var instance: SearchDatabase
        @JvmStatic
        @Synchronized
        fun getDatabase(context: Context): SearchDatabase {
            instance = Room.databaseBuilder(context.applicationContext,
                    SearchDatabase::class.java, DATABASE_NAME
            )
                    .fallbackToDestructiveMigration()
                    .build()
            return instance
        }
    }
}